sql/core/src/test/scala/org/apache/spark/sql/execution/SQLExecutionSuite.scala

```diff
@@ -119,6 +125,39 @@ class SQLExecutionSuite extends SparkFunSuite {
 
     spark.stop()
   }
+
+  test("SPARK-32813: Table scan should work in different thread") {
+    val executor1 = Executors.newSingleThreadExecutor()
+    val executor2 = Executors.newSingleThreadExecutor()
+    var session: SparkSession = null
+    SparkSession.cleanupAnyExistingSession()
+
+    withTempDir { tempDir =>
+      try {
+        val tablePath = tempDir.toString + "/table"
```

Comment:
nit: why don't we use `tempDir` directly?

Reply:
will complain ... dir was already existing.

Reply:
oh, can we try `withTempPath`?

Reply:
https://github.com/apache/spark/pull/29667#discussion_r484625824 :-)